okio/src/main/java/okio/Buffer.java

```diff
@@ -529,6 +529,41 @@ public byte getByte(long pos) {
     return new ByteString(readByteArray(byteCount));
   }
 
+  @Override public int select(List<ByteString> byteStrings) {
+    Segment s = head;
+    if (s == null) return byteStrings.indexOf(ByteString.EMPTY);
```

Comment:
I guess it comes down to whether `"".starts with("")` is true or not. I'd prefer to not make this a special case, but when the Buffer is empty it doesn't have any segments so we can't use the regular flow.